src/instantsend/instantsend.h

```diff
@@ -148,8 +128,16 @@ class CInstantSendManager final : public instantsend::InstantSendSignerParent
     bool IsWaitingForTx(const uint256& txHash) const EXCLUSIVE_LOCKS_REQUIRED(!cs_pendingLocks);
     instantsend::InstantSendLockPtr GetConflictingLock(const CTransaction& tx) const override;
 
-    [[nodiscard]] MessageProcessingResult ProcessMessage(NodeId from, std::string_view msg_type, CDataStream& vRecv)
+    /* Helpers for communications between CInstantSendManager & NetInstantSend */
+    // This helper returns up to 32 pending locks and remove them from queue of pending
+    [[nodiscard]] instantsend::PendingState FetchPendingLocks() EXCLUSIVE_LOCKS_REQUIRED(!cs_pendingLocks);
+    void EnqueueInstantSendLock(NodeId from, const uint256& hash, const std::shared_ptr<instantsend::InstantSendLock>& islock)
         EXCLUSIVE_LOCKS_REQUIRED(!cs_pendingLocks);
+    [[nodiscard]] std::vector<CTransactionRef> PrepareTxToRetry() EXCLUSIVE_LOCKS_REQUIRED(!cs_nonLocked, !cs_pendingRetry);
+    CSigningManager& Sigman() { return sigman; }
+    [[nodiscard]] std::variant<uint256, CTransactionRef, std::monostate> ProcessInstantSendLock(
+        NodeId from, const uint256& hash, const instantsend::InstantSendLockPtr& islock)
+        EXCLUSIVE_LOCKS_REQUIRED(!cs_nonLocked, !cs_pendingLocks, !cs_pendingRetry);
```

Comment:
why are we returning a variant

Reply:
because we need to return CTransaction OR uint256 OR nothing.

If it will be (CTransaction OR nothing) I will use std::optional
But there's 3 options. So, variant is a choice.